errors.go

```diff
@@ -7,7 +7,7 @@ import (
 
 var (
 	ErrInvalidKey                = errors.New("key is invalid")
-	ErrInvalidKeyType            = errors.New("key is of invalid type")
+	ErrInvalidKeyType            = errors.New("key is of invalid go type (should be []byte())")
```

Comment:
This is problematic because it is used for all signing methods and the invalid type can be different ones, depending on the signing method. The only way to make this better is to add additional detail in the signing methods where the error is used.

Reply:
Would you recommend to add a new Err into the Err list like so:
```
var (
	ErrInvalidKey                = errors.New("key is invalid")
	ErrInvalidKeyType            = errors.New("key is of invalid go type")
	...
 	ErrHmacKeyType               = errors.New("hmac expects []byte")
)
```
and then using it like so:
```
if !ok {
        return newError("", ErrInvalidKeyType, ErrHmacKeyType)
}
```

Or rather like this without creating a new Err in the list:
```
if !ok {
	return newError("hmac expects []byte", ErrInvalidKeyType)
}
```

Reply:
Then I would add this for the rest of the Signing Methods as well, that should work then right?

Reply:
I think I would prefer the latter, I think its enough for people to check on the more general `ErrInvalidKeyType`, the specifics are fine as a string within `newError`.